### ISSUE TITLE
issue with static destination

### ISSUE THREAD

**Issue by Renand** [rip747/Uploadable-Files#5]
Hi tony, tanks for your Help, When I update the plugin to 0.5 I find a new message :

Element C:\Inetpub\wwwroot\SITES\adminOnWheels\files\temp\ is undefined in a Java object of type class coldfusion.runtime.TemplateProxy.

The error occurred in C:\Inetpub\wwwroot\SITES\adminOnWheels\plugins\uploadablefiles\uploadablefiles.cfc: line 193
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\plugins\uploadablefiles\uploadablefiles.cfc: line 77
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\global\cfml.cfm: line 174
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\model\callbacks.cfm: line 291
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\model\crud.cfm: line 830
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\global\cfml.cfm: line 174
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\model\transactions.cfm: line 53
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\model\crud.cfm: line 816
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\model\crud.cfm: line 896
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\controllers\Tissues.cfc: line 225
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\global\cfml.cfm: line 174
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\controller\processing.cfm: line 85
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\controller\processing.cfm: line 60
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\dispatch\request.cfm: line 166
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\index.cfm: line 1
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\rewrite.cfm: line 1
Called from C:\Inetpub\wwwroot\SITES\adminOnWheels\wheels\events\onrequest.cfm: line 1

I will try to understand why, but my first question is :
Actually I use a temp folder defined in my model like this : destination=expandPath(application.folders.uploadTemp) // /files/temp/
And I move my file with atfersave() medhod in other destination, like this one /images/CONTROLLER_NAME/OBJECT_ID/renamed_file_name.jpg (UUID in fact)

I saw in plugin code (_getUFProperty) a way to defined a dynamic destination folder, but I don't know if it's possible to use it if I want to use ID in path.

Thanks

**Comment by rip747** [comment]
To answer you question about the dyanmic destination, you won't be able to
do that on a create.
On Oct 10, 2011 9:18 AM, "Renand" <
reply@reply.github.com>
wrote:

**Comment by Renand** [comment]
Il will try to :)

**Comment by rip747** [comment]
fixed in 0.6.

**Comment by Renand** [comment]
Sorry Tony :( but another issue arrive with this update.

When I upload a file for exemple for an "image" property like that :

MODEL :
uploadableFile(property="image", message="#$('error.imageUpload')#", removeOnDelete=false, destination=expandPath(application.folders.uploadTemp));

I use removeOnDelete=false because i'm not be able to store image path in DB, and I don't want it in the DB.

CONTROLLER :
product.uploadableFileFieldName(image="product[image]");

the file goes on the server, but the image property was not field with image fillename. And when I dump my object,  image is not present and _uf_original_image is empty.

 In the 0.4 version this.image return image name.

**Comment by rip747** [comment]
is the image property a column on the table? i don't see why you would need to use the uploadableFileFieldName() method if it is. post the schema and model component for the table that you are using the plugin with to a gist and paste a link here. if you can also include the controller and view code, that would also be helpful.

**Comment by Renand** [comment]
Le 12 oct. 2011 à 12:17, tony petruzzi a écrit :

> is the image property a column on the table?
> Yes
> 
> i don't see why you would need to use the uploadableFileFieldName()method if it is.
> Ok, I don't understand it is an option, thanks for this information
> 
> post the schema and model component for the table that you are using the plugin with to a gist and paste a link here. if you can also include the controller and view code, that would also be helpful.
> Ok I will try, I'm very newbie, can I put it in zip file ?

Thanks very much Tony for your help and for your work on wheels.

**Comment by rip747** [comment]
> Ok, I don't understand it is an option, thanks for this information

there isn't any harm in using it, but it's really there where you have when you have a fieldname that doesn't follow the wheels convention. if your using the filefield() method, all should be good. i just thought that since you were using it that the image property wasn't a column on the table the model is associated with.

> Ok I will try, I'm very newbie, can I put it in zip file ?

i don't think you can post it as a zip file to issues (i could be wrong). if you don't know how to use gists, no worries, just paste the code into a reply to this issue.

> Thanks very much Tony for your help and for your work on wheels.

your welcome :)

**Comment by Renand** [comment]
I think is ok with gists :

https://gist.github.com/1281063

**Comment by Renand** [comment]
Sorry I need to remove the SQL dump because the datas it's not mine.
Could I send you the dump by email ?

**Comment by rip747** [comment]
awesome! looking at the code, i really don't see anything wrong with it. i'm wondering though if maybe something is up with the expandpath statement in the destination:

```
uploadableFile(property="image", message="#$('error.imageUpload')#", removeOnDelete=false, destination=expandPath(application.folders.uploadTemp));
```

just for kicks, try wrapping the destination in quotes and pound signs:

```
uploadableFile(property="image", message="#$('error.imageUpload')#", removeOnDelete=false, destination="#expandPath(application.folders.uploadTemp)#");
```

i'll look more into this tonight and test some things out when i get home. also PM me your email address and i will send you 0.7 directly so i can get your confirmation that the issue is resolved before i release yet another fucked up version to the world :)

**Comment by Renand** [comment]
thanks tony.

**Comment by rip747** [comment]
fixed in 0.7
